File: ebs/lib/model/SalesOrder.dart
class SalesOrder {
  final int? id;
  final int? salesOrderCode;
  final int? customerCode;
  final String? customerName;
  final DateTime? orderDate;
  final DateTime? deliveryDate;
  final int? productCode;
  final String? productName;
  final double? unitPrice;
  final int? salesQuantity;
  final double? totalAmount;
  final String? paymentStatus;
  final String? status;

  SalesOrder({
    this.id,
    this.salesOrderCode,
    this.customerCode,
    this.customerName,
    this.orderDate,
    this.deliveryDate,
    this.productCode,
    this.productName,
    this.unitPrice,
    this.salesQuantity,
    this.totalAmount,
    this.paymentStatus,
    this.status,
  });

  factory SalesOrder.fromJson(Map<String, dynamic> json) {
    return SalesOrder(
      id: json['id'],
      salesOrderCode: json['salesOrderCode'],
      customerCode: json['customerCode'],
      customerName: json['customerName'],
      orderDate: json['orderDate'] != null ? DateTime.parse(json['orderDate']) : null,
      deliveryDate: json['deliveryDate'] != null ? DateTime.parse(json['deliveryDate']) : null,
      productCode: json['productCode'],
      productName: json['productName'],
      unitPrice: json['unitPrice'] != null ? (json['unitPrice'] as num).toDouble() : null,
      salesQuantity: json['salesQuantity'],
      totalAmount: json['totalAmount'] != null ? (json['totalAmount'] as num).toDouble() : null,
      paymentStatus: json['paymentStatus'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salesOrderCode': salesOrderCode,
      'customerCode': customerCode,
      'customerName': customerName,
      'orderDate': orderDate?.toIso8601String(),
      'deliveryDate': deliveryDate?.toIso8601String(),
      'productCode': productCode,
      'productName': productName,
      'unitPrice': unitPrice,
      'salesQuantity': salesQuantity,
      'totalAmount': totalAmount,
      'paymentStatus': paymentStatus,
      'status': status,
    };
  }
}

File: ebs-springboot/src/main/java/com/erp/model/SalesOrder.java
package com.erp.model;

import javax.persistence.*;
import java.util.Date;

@Entity(name = "sales_order")
@Table(name = "sales_order")
public class SalesOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "sales_order_code")
    private Long salesOrderCode;

    @Column(name = "customer_code")
    private Long customerCode;

    @Column(name = "customer_name")
    private String customerName;

    @Column(name = "order_date")
    @Temporal(TemporalType.DATE)
    private Date orderDate;

    @Column(name = "delivery_date")
    @Temporal(TemporalType.DATE)
    private Date deliveryDate;

    @Column(name = "product_code")
    private Long productCode;

    @Column(name = "product_name")
    private String productName;

    @Column(name = "unit_price")
    private Double unitPrice;

    @Column(name = "sales_quantity")
    private Integer salesQuantity;

    @Column(name = "total_amount")
    private Double totalAmount;

    @Column(name = "payment_status")
    private String paymentStatus;

    @Column(name = "status")
    private String status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getSalesOrderCode() {
        return salesOrderCode;
    }

    public void setSalesOrderCode(Long salesOrderCode) {
        this.salesOrderCode = salesOrderCode;
    }

    public Long getCustomerCode() {
        return customerCode;
    }

    public void setCustomerCode(Long customerCode) {
        this.customerCode = customerCode;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public Date getOrderDate() {
        return orderDate;
    }

    public void setOrderDate(Date orderDate) {
        this.orderDate = orderDate;
    }

    public Date getDeliveryDate() {
        return deliveryDate;
    }

    public void setDeliveryDate(Date deliveryDate) {
        this.deliveryDate = deliveryDate;
    }

    public Long getProductCode() {
        return productCode;
    }

    public void setProductCode(Long productCode) {
        this.productCode = productCode;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public Double getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(Double unitPrice) {
        this.unitPrice = unitPrice;
    }

    public Integer getSalesQuantity() {
        return salesQuantity;
    }

    public void setSalesQuantity(Integer salesQuantity) {
        this.salesQuantity = salesQuantity;
    }

    public Double getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(Double totalAmount) {
        this.totalAmount = totalAmount;
    }

    public String getPaymentStatus() {
        return paymentStatus;
    }

    public void setPaymentStatus(String paymentStatus) {
        this.paymentStatus = paymentStatus;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
    
}
